ElectronClient/gui/NoteEditor/NoteBody/AceEditor/AceEditor.tsx

```diff
@@ -5,14 +5,15 @@ import { useState, useEffect, useRef, forwardRef, useCallback, useImperativeHand
 import { EditorCommand, NoteBodyEditorProps } from '../../utils/types';
```

Comment:
Any reason for the changes on Ace Editor related files? The app can handle multiple text editors, which makes it much easier to migrate to other editors. In NoteEditor.tsx, you could simply import your CodeMirror component, then if props.bodyEditor === "CodeMirror", load it.

So I guess instead of changing the AceEditor files, you'd just copy what you've done so far to NoteBody/CodeMirror and revert the changes on the AceEditor files. If you need some more info on how to get it working let me know.

Reply:
Basically just laziness, I didn't know the new aspects of the code very well and I didn't want to waste time figuring out how to support a third editor before I even had one working. 
I was initially planning on doing exactly what you suggested before making this public but decided it was better to get some visibility first. I'll definitely make those changes before taking this out of draft.